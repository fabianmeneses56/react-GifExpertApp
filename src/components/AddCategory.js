import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setcategories }) => {
  const [inputValue, setinputValue] = useState("");

  const handleInputChange = (e) => {
    setinputValue(e.target.value);

    console.log('handleInput llamado')
  };

  const handleSubmit = (e) => {
    /* prevenir que la pagina se actualice por navegador */
    e.preventDefault();
    console.log('handle submit',inputValue);
    /* validacion de que hallna mas de 2 letras */
    if (inputValue.trim().length > 2) {
      setcategories((cats) => [inputValue, ...cats]);
      setinputValue("");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <p>{ inputValue }</p>
      <input type="text" value={inputValue} onChange={handleInputChange} />
    </form>
  );
};

AddCategory.propTypes = {
  setcategories: PropTypes.func.isRequired,
};
