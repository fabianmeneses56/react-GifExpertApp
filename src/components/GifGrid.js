import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";
import PropTypes from 'prop-types'

// EN ESTE SE IMPLEMENTO UN CUSTOM HOOK QUE HACE EL TRABAJO PESADO
export const GifGrid = ({ category }) => {
                    /*   const [images, setimages] = useState([]); */

                    /* use effect es como el component didmount */
  const { data:images,loading } = useFetchGifs( category );
                    /*  useEffect(() => {
                        //getGifs(); -> como ya no tengo el getgifs lo importo
                        getGifs(category).then(setimages);
                        //.then(imgs=>setimages(imgs))
                    }, [category]); */

                    /*  getGifs(); */

  return (
    <>
      <h3 className="animate__animated animate__fadeIn">{category}</h3>

      {loading && <p className="animate__animated animate__flash">cargando</p>}
     
      <div className="card-grid">
        <ol className="ol-grid">
          {images.map((img) => (
            <GifGridItem
              key={img.id}
              {...img}
              //img={img}
            />
          ))}
        </ol>
      </div>
    </>
  );
};
GifGrid.propTypes={
  category:PropTypes.string.isRequired
}