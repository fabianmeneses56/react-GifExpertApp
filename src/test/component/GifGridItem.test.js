import React from 'react';
import {shallow}  from 'enzyme';
import  {GifGridItem} from '../../components/GifGridItem';


describe('Pruebas en <GifGridItem />', () => {

    const title = "un titulo";
    const url = "algo";// estos argumentos se tienen que poner para que pase la prueba ya que se hicieron los propsTypes para que sean requeridos
    const wrapper = shallow(<GifGridItem title={ title } url={ url }/>)
    
    test('debe mostrar el componente correctamente', () => {
        
        expect( wrapper ).toMatchSnapshot();
    })
    
    test('debe tener un parrafo con el cliente', () => {
        
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );
    })
    
    test('debe tener la imagen igual al url y alt de los props ', () => {
        
        const img = wrapper.find('img');//<- etiqueta
        console.log( img.props().src)
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
        /* console.log( img.html()) */
    });

    test('debe tener animate:fadeIn', () => {
        
        const div= wrapper.find('div');
        const className = div.prop('className');
        
        expect( className.includes('animate__fadeIn') ).toBe( true );
    })
    
    
    
})
