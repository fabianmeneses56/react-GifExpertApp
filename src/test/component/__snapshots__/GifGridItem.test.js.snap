// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="un titulo"
    src="algo"
  />
  <p>
    un titulo
  </p>
</div>
`;
